[MASWE-0009] Add Weak Cryptographic Key Generation (by appknox) (#2849)
OWASP/owasp-mastg: demos/android/MASVS-CRYPTO/MASTG-DEMO-0012/MASTG-DEMO-0012.md

---
platform: android
title: Weak Cryptographic Key Generation

> tools: [semgrep]

Reviewer: ```suggestion
tools: [MASTG-TOOL-0105]
```

> tools: [MASTG-TOOL-0105]

Reviewer 2: This can be removed. Please include `@ MASTG-TOOL-0109` below in the text.

code: [java]
id: MASTG-DEMO-0012
test: MASTG-TEST-0208
---

### Sample

> {{ MastgTest_keylength.java }}

Reviewer: ```suggestion
{{ MastgTest.java }}
```

### Steps

Let's run our semgrep rule against the sample code.

{{ ../../../../rules/mastg-android-weak-crypto-key-generation.yml }}

{{ run.sh }}

### Observation

The rule has identified all the location in the code file where weak cipher key is used.

{{ output.txt }}

### Evaluation

Review each of the reported instances.

- Line 2 has initialized the RSA key size of 1024 bits.

> - Line 5 using `keyGen.init(64)` initializer creates a symmetric key with a size of 64 bits (8 bytes).

Reviewer: ```suggestion
- Line 5 is using `keyGen.init(64)` which creates a symmetric key with a size of 64 bits (8 bytes).
```
